frontend/src/dataUtilities.ts

```diff
@@ -475,3 +504,28 @@ export const transformPaymentData = (data: PaymentData) => [
     recipientId: data.payment.recipient_id,
   },
 ];
+
+interface AssetResponseData {
+  json: AssetReport;
+}
+
+export const transformAssetsData = (data: AssetResponseData) => {
+  const assetItems = data.json.items;
+  let final: DataItem[] = [];
+  assetItems.forEach((item) => {
+    final = final.concat(
```

Comment:
could this be more immutable and more concise? also where is this `json` key coming from? none of the others have the data nested under `.json`

```tsx
export const transformAssetsData = (data: AssetResponseData): Array<DataItem> =>
  data.json.items.map((item) =>
    item.accounts.map((account) => ({
      account: account.name,
      balance:
        formatter.format(account.balances.available!) ||
        formatter.format(account.balances.current!),
      transactions: account.transactions!.length,
      daysAvailable: account.days_available!,
    }))
  );
```

Reply:
changed to flatMap